### PR TITLE
net/rpmsg: add multi-iov support

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Add multi-iov support for rpmsg sockets

### CODE REVIEW

**Comment by gustavonihei** [comment]
I've never worked with the RPMSG subsystem, I'd need some time to understand these changes.

**Comment by xiaoxiang781216** [comment]
Sure, the key change is support send iovec in one batch.

File: net/rpmsg/rpmsg_sockif.c [hunk 1]
```diff
@@ -991,7 +1021,7 @@ static ssize_t rpmsg_socket_send_single(FAR struct socket *psock,
   if (total > ipcsize)
     {
       total = ipcsize;
-      len   = ipcsize - sizeof(*msg) + sizeof(uint32_t);
+      len = ipcsize - sizeof(*msg) - sizeof(uint32_t);
```
Comment (gustavonihei) [inline net/rpmsg/rpmsg_sockif.c:1024]: This seems to be a bugfix that is not strictly related to the new feature being added in this PR.
If so, please add it to a dedicated commit.

File: net/rpmsg/rpmsg_sockif.c [hunk 2]
```diff
@@ -862,13 +862,26 @@ static int rpmsg_socket_poll(FAR struct socket *psock,
   return ret;
 }
 
+static uint32_t rpmsg_socket_get_iovlen(FAR const struct iovec *buf,
+                                       size_t iovcnt)
```
Comment (gustavonihei) [inline net/rpmsg/rpmsg_sockif.c:866]:
```suggestion
                                        size_t iovcnt)
```
nit: alignment.

Comment (xiaoxiang781216) [reply net/rpmsg/rpmsg_sockif.c:866]: Done.

File: net/rpmsg/rpmsg_sockif.c (hunk 1 above)
Comment (xiaoxiang781216) [reply net/rpmsg/rpmsg_sockif.c:1024]: The original code looks good, I drop this change directly.

**Review by gustavonihei — APPROVED** [review]
LGTM. From the quick look I've taken at the RPMsg protocol, I think these changes seem okay.
